server/sql.go

```diff
@@ -992,3 +1027,42 @@ func isMySQLExcludeDatabase(database string) bool {
 	}
 	return true
 }
+
+func (s *Server) hasDatabaseAccessRights(ctx context.Context, principalID int, database *api.Database) (bool, error) {
+	// Only project member can access database.
+	if !api.HasActiveProjectMembership(principalID, database.Project) {
+		return false, nil
+	}
+
+	// calculate the effective policy.
+	databasePolicy, inheritFromEnvironment, err := s.store.GetNormalAccessControlPolicy(ctx, api.PolicyResourceTypeDatabase, database.ID)
+	if err != nil {
+		return false, err
+	}
+
+	environmentPolicy, _, err := s.store.GetNormalAccessControlPolicy(ctx, api.PolicyResourceTypeEnvironment, database.Instance.EnvironmentID)
+	if err != nil {
+		return false, err
+	}
+
+	if !inheritFromEnvironment {
+		// Use database policy.
+		return databasePolicy != nil, nil
```

Comment:
We use default policy I think and we probably should. Use databasePolicy != nil && len(databasePolicy.DisallowRuleList) == 0.

Reply:
updated.